src/test/scala/mesosphere/marathon/core/task/TaskTest.scala

```diff
@@ -247,4 +246,68 @@ class TaskTest extends UnitTest with Inside {
     }
   }
 
+  val marathon_1_4_9_ephemeral_app_id = "sleeper.7dcb9c7f-e01d-11e9-80dd-2ae4d7ed9076"
+  val marathon_1_4_9_ephemeral_pod_id = "dev_sleeper.instance-71471cee-e01d-11e9-80dd-2ae4d7ed9076.sleep1"
+  val marathon_1_4_9_resident_app_id = "dev_resident-sleep.d15865ae-e01f-11e9-8fd4-2ae4d7ed9076.2"
+  "task ID parsing" should {
```

Comment:
Awesome. I just wanted to point you to `Task.Id.parse`. Thanks for going the extra mile!